index.ts

```diff
@@ -1,35 +1,65 @@
 export class OatyArray {
-  private _original: object[]
-  private _keys: string[]
-  private _transposed: object = {}
+  private keys: string[] | undefined
+  private original: object[] = []
+  private transposed: { [key: string]: any } = {}
+
+  constructor(options?: { data?: object[], keys?: string[] }) {
+    if (options !== undefined) {
+      if (options.data !== undefined) { this.push(...options.data) }
+      if (options.keys !== undefined) { this.keys = options.keys }
+    }
+  }
 
   get length(): number {
-    return this._original.length
+    return this.original.length
   }
 
-  constructor(input: object[], keys: string[]) {
-    this._original = input.map((x) => Object.assign({}, x))
-    this._keys = keys
+  public getKeys(): string[] | undefined {
+    return this.keys
+  }
 
-    this._transposed = transpose(this._keys, this._original)
+  public getOriginal(): object[] {
+    return this.original
   }
 
-  public get(keyName: string, keyValue: string): object[] {
-    return this._transposed[keyName][keyValue]
+  public getTransposed(): object[] {
+    return this.original
   }
 
-  public push(...items: object[]): number {
-    this._transposed = transpose(this._keys, items, this._transposed)
-    return this._original.push(...items)
+  public get(keyName: string, keyValue: string): object[] | undefined {

```

Comment:
Also, should we rename this to `find` to be more consistent with other things?

Reply:
Might be misleading - `Array` `find` takes a function parameter, `OatyArray` `get` doesn't